docusaurus/docs/iOS/customization/localization.md

```diff
@@ -0,0 +1,50 @@
+---
+title: Translation and i10n
+---
+
+The SDK offers you in-house localization for all strings contained in UI components. For our internal localization, we provide `strings` file for each language that we translate. For advanced plural localization cases (typing indicator showing how many people are typing or how many responses are in thread), we use `stringsdict` with `NSStringPluralRuleType`. 
+
+## The Change of language on the device.
+
+Language for the StreamChat is changed together language of the device or simulator. See [how to change device language](https://support.apple.com/en-us/HT204031).
+There are however some workarounds how you can enforce the application to use only specific languages or let the users choose.
+
+## Adding new language which is not present in StreamChat
+
+StreamChatUI uses `strings` and `stringsdict` files to handle localize the framework.
+`Appearance` structure contains closure `localizationProvider`, which is responsible for deriving `localizedString` from exact bundle. 
+By default, this bundle is the  `StreamChatUI` framework bundle, but you can customize it to anything you need in 3 steps:
+
+1. If you don't have yet your `strings` or `stringsdict` files, you need to create those new files in your project/target, preferably `Localizable.strings` and `Localizable.stringsdict`.
+2. When you have those files present, you need to [add new language to thole project](https://developer.apple.com/forums/thread/97252?answerId=295557022#295557022).
+3. You need to do is to copy available `StreamChatUI` localization keys into your `strings` and `stringsdict` files.
+4. Once you have those files in your application, you need to set the `localizationProvider` to provide your `AppBundle` instead of `StreamChatUI` frameworks like this: 
+```swift
+Appearance.default.localizationProvider = { key, table in
+    // For most cases, you will probably use Bundle.main, unless your strings are in different Bundle.
+    // Note: `table` is equivalent to filename of the `Localization.strings` file without file extension. 
+    // the name for `strings` and `stringsdict` should be same!
+    Bundle.main.localizedString(forKey: key, value: nil, table: table)
+}
+```
+5. Now everything that has left is to implement your  `strings` and `stringsdict`  files for different languages or even customize the default strings for your own.
+
+This step is the same for overriding existing languages, because there is no user-friendly way to override just some keys from the `strings` file as the Localization is getting loaded as whole file.
```

Comment:
Do we have the steps for overriding existing languages recorded in the docs? If yes, link to that page to make this easier for users to navigate.

Reply:
Nope, because it's the same, in the cocoa world, you cannot simply override to localization keys only without doing some swizzling system calls and hacks.